lib/devise/models/recoverable.rb

```diff
@@ -32,7 +32,7 @@ def self.required_fields(klass)
 
       included do
         before_save do
-          if email_changed? || encrypted_password_changed?
+          if persisted? && (email_changed? || encrypted_password_changed?)
```

Comment:
What do you think of wrapping this up into an instance method with a piece of documentation explaining the semantics of all these checks at once? I think this will help out whenever we need to revisit this later.